extensions/wikia/EmbeddableDiscussions/scripts/DiscussionsSettingsModal.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Modal containing settings for the Discussions Module
+ */
+define('DiscussionsSettingsModal',
+	['jquery', 'wikia.loader', 'mw', 'wikia.mustache', 'wikia.tracker', 'wikia.nirvana'],
+	function ($, loader, mw, mustache, tracker, nirvana) {
+		'use strict';
+		var modalConfig = {
+				vars: {
+					id: 'DiscussionsSettingsModal',
+					classes: ['embeddable-discussions-settings-modal'],
+					size: 'small'
+				}
+			},
+			track = tracker.buildTrackingFunction({
+				action: tracker.ACTIONS.CLICK,
+				category: 'community-page-discussions-settings-modal',
+				trackingMethod: 'analytics'
+			}),
+			config;
+
+		function saveSettings() {
+			var $selectedCategories = $('#embeddable-discussions-settingsmodal-categories :selected');
+
+			// Update selected categories before sending request
+			config.selectedCategories = [];
+
+			if (!$('#embeddable-discussions-settingsmodal-filter-by-all-checkbox').is(':checked')) {
+				// All not checked, populate with selected items in list
+				$selectedCategories.each(function() {
+					config.selectedCategories.push({ id: $(this).val() });
+				});
+			}
+
+			nirvana.sendRequest({
+				controller: 'EmbeddableDiscussionsApi',
+				method: 'saveCommunityPageSettings',
+				type: 'post',
+				format: 'json',
+				data: config,
+			}).then(function(data) {
+				console.log(data);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)